hie-plugin-api/Haskell/Ide/Engine/Cradle.hs

```diff
@@ -219,29 +222,29 @@ findCabalHelperEntryPoint fp = do
   let supportedProjs = filter (\x -> supported x isStackInstalled isCabalInstalled) allProjs
   debugm $ "These projects have the build tools installed: " ++ show (map (\(Ex x) -> show x) supportedProjs)
 
-  case filter (\p -> isCabalNewProject p || isStackProject p) supportedProjs of
+  case filter (\p -> isCabalV2FileProject p || isStackProject p) supportedProjs of
     (x:_) -> return $ Just x
-    []    -> case filter isCabalOldProject supportedProjs of
+    []    -> case filter isCabalProject supportedProjs of
       (x:_) -> return $ Just x
       []    -> return Nothing
     where
-      supported :: (Ex ProjLoc) -> Bool -> Bool -> Bool
-      supported (Ex ProjLocStackYaml {}) stackInstalled _ = stackInstalled
-      supported (Ex ProjLocV2Dir {}) _ cabalInstalled = cabalInstalled
-      supported (Ex ProjLocV2File {}) _ cabalInstalled = cabalInstalled
-      supported (Ex ProjLocV1Dir {}) _ cabalInstalled = cabalInstalled
-      supported (Ex ProjLocV1CabalFile {}) _ cabalInstalled = cabalInstalled
+      supported :: Ex ProjLoc -> Bool -> Bool -> Bool
+      supported proj stackInstalled cabalInstalled
+        | isCabalProject proj = cabalInstalled
+        | isStackProject proj = stackInstalled
+        | otherwise = False
```

Comment:
I thought being explicit is better than implicit. It may happen that c-h starts to support more projects then just cabal and stack and we dont want to misidentify them as stack projects.
Actually, is the old code better since this would turn such an event into a compile-time error?

Reply:
There is something to be said for exhaustive matching, because the compiler can then warn about missing matches.